packages/ui/src/components/SelectInputV2/SelectBar.tsx

```diff
@@ -0,0 +1,337 @@
+  disabled: boolean
+  readOnly: boolean
+  selectedValues: (OptionType | undefined)[]
+  setSelectedValues: Dispatch<SetStateAction<(OptionType | undefined)[]>>
+  onChange?: (value: (string | undefined)[]) => void
+  overflowed: boolean
+  overflowAmount: number
+}
+const StateStack = styled(Stack)`
+  padding-right: ${({ theme }) => theme.space['2']};
+  right: 0px;
+  display: flex;
+`
+
+const StackTags = styled(Stack, {
+  shouldForwardProp: prop => !['width'].includes(prop),
+})<{ width: number | undefined }>`
+  max-width: ${({ width }) => width}px;
+  overflow: scroll;
+`
+const StyledInputWrapper = styled(Stack, {
+  shouldForwardProp: prop =>
+    !['size', 'isDropdownVisible', 'state', 'openable'].includes(prop),
+})<StyledInputWrapperProps>`
+  display: flex;
+  height: ${({ size }) => INPUT_SIZE_HEIGHT[size]}px;
+  padding: ${({ theme }) => theme.space[1]};
+  padding-right: 0;
+  padding-left: ${({ theme }) => theme.space[2]};
+
+  background: ${({ theme }) => theme.colors.neutral.background};
+  border: 1px solid ${({ theme, state }) => theme.colors[state].border};
+  border-radius: ${({ theme }) => theme.radii.default};
+
+  &:hover,
+  :focus {
+    border-color: ${({ theme }) => theme.colors.primary.borderHover};
+    outline: none;
+  }
+  &[data-readonly='true'] {
+    background: ${({ theme }) => theme.colors.neutral.backgroundWeak};
+    border-color: ${({ theme }) => theme.colors.neutral.border};
+  }
+
+  &[data-disabled='true'] {
+    background: ${({ theme }) => theme.colors.neutral.backgroundDisabled};
+    border-color: ${({ theme }) => theme.colors.neutral.borderDisabled};
+  }
+
+  &:not([data-disabled='true']):not([data-readonly]):hover {
+    border-color: ${({ theme }) => theme.colors.primary.border};
+  }
+
+  &:active {
+    box-shadow: ${({ theme, openable }) =>
+      openable ? theme.shadows.focusPrimary : 'none'};
+  }
+
+  ${({ theme, isDropdownVisible }) =>
+    !isDropdownVisible
+      ? ''
+      : `
+          box-shadow: ${theme.shadows.focusPrimary};
+          border-color: ${theme.colors.primary.borderHover};
+          }
+        `}
+`
+const CustomTag = styled(Tag)`
+  height: fit-content;
+  width: fit-content;
+`
+
+const StyledPlaceholder = styled(Text)`
+  color: ${({ theme }) => theme.colors.neutral.textWeak};
+  text-size: ${({ theme }) => theme.typography.body.fontSize};
+  display: flex;
+  flex: 1;
+  align-self: center;
+`
+
+const DisplayValues = ({
+  multiselect,
+  refTag,
+  width,
+  nonOverflowedValues,
+  disabled,
+  readOnly,
+  selectedValues,
+  setSelectedValues,
+  onChange,
+  overflowed,
+  overflowAmount,
+}: DisplayValuesProps) =>
+  multiselect ? (
+    <StackTags
+      direction="row"
+      gap="1"
+      wrap="nowrap"
+      ref={refTag}
+      width={width}
+      alignItems="center"
+    >
+      {nonOverflowedValues.map((option, index) => (
+        <CustomTag
+          data-testid="selected-options-tags"
+          key={option ? option.value : index}
```

Comment:
Won't it be risky that two options have the same value?

Reply:
Shouldn't the values be unique ?

Reply:
good question haha I'd say yes but technically the user could set 2 same values